lib/src/state/design.dart

```diff
@@ -506,6 +506,22 @@ abstract class Design with UnusedFields implements Built<Design, DesignBuilder>,
     return domain_mismatches_builtmap_builder.build();
   }
 
+  @memoized
+  BuiltMap<Domain, BuiltList<Mismatch>> get insertion_deletion_mismatches_map {
+    var insertion_deletion_mismatches_map_builder = MapBuilder<Domain, ListBuilder<Mismatch>>();
+    for (Strand strand in this.strands) {
+      for (Domain domain in strand.domains) {
+        insertion_deletion_mismatches_map_builder[domain] =
+            this._find_mismatches_on_substrand(domain, only_insertion_deletion: true);
+      }
+    }
+    var insertion_deletion_mismatches_builtmap_builder = MapBuilder<Domain, BuiltList<Mismatch>>();
```

Comment:
This line seems strange to me. It doesn't look like there's any need to call `build()` on the `MapBuilder` named `insertion_deletion_mismatches_map_builder`.

Actually, in general it's good to imitate existing code (and maybe some nearby code looks like this), but in this case I'd do this differently. In particular, the built library provides [extension methods](https://dart.dev/guides/language/extension-methods#:~:text=Extension%20methods%2C%20introduced%20in%20Dart,extension%20methods%20alongside%20regular%20methods.) so that you can call `build()` directly on native collections such as `List`'s and `Map`'s. I find that more readable. So I'd implement this something like (haven't tested this code though):

```dart
var insertion_deletion_mismatches_map = Map<Domain, List<Mismatch>>();
for (Strand strand in this.strands) {
  for (Domain domain in strand.domains) {
    insertion_deletion_mismatches_map[domain] =
        this._find_mismatches_on_substrand(domain, only_insertion_deletion: true);
  }
}
var insertion_deletion_mismatches_half_built_map = Map<Domain, BuiltList<Mismatch>>();
insertion_deletion_mismatches_map.forEach((domain, mismatches)) {
  insertion_deletion_mismatches_half_built_map[domain] = mismatches.build();
}
return insertion_deletion_mismatches_half_built_map.build();
```

Which is a bit cleaner to read, and doesn't do the unnecessary `build()` on the first data structure.